inference-engine/thirdparty/clDNN/kernel_selector/core/cl_kernels/pooling_gpu_ref.cl

```diff
@@ -227,7 +258,15 @@ KERNEL(pooling_gpu)(
                 uint input_idx = INPUT0_GET_INDEX(b, f, offset_y + j, offset_x + i);
                 result = FUNC_CALL(apply_pooling)(result, TO_ACCUMULATOR_TYPE(input[input_idx]));
     #else
-                result = FUNC_CALL(apply_pooling)(result, TO_ACCUMULATOR_TYPE(input[input_idx]));
+                #ifdef SECOND_OUTPUT_TYPE
+                    if (input[input_idx] > result)
```

Comment:
Minimize reading from the global memory. One of solutions would be to have an additional variable which keeps previous result value.